src/image.js

```diff
@@ -0,0 +1,121 @@
+require("babel-polyfill");
+
+import gm from "gm";
+import log from "./log";
+import uuid from "uuid";
+
+const im = gm.subClass({imageMagick: true});
+
+// Wrap these calls in promises so we can use async/await
+const promiseWrite = (client, file) => {
+  return new Promise((resolve, reject) => client.write(file, err => err ? reject(err) : resolve()));
+};
+const promiseSize = (client) => {
+  return new Promise((resolve, reject) => client.size((err, data) => err ? reject(err) : resolve(data)));
+};
+
+// Interlacing for png isn't efficient (both in filesize as render performance), so we only do it for jpg
+const interlace = async(client, params) => {
```

Comment:
Another pro-tip: async also works for regular functions: `async function interlace(client, params) {}`.